packages/vue/src/auth/entitlements.ts

```diff
@@ -9,36 +9,85 @@ import {
   AuthState,
 } from '@frontegg/redux-store';
 
+import { FeatureFlags, USE_ENTITLEMENTS_V2_ENDPOINT_FF } from '@frontegg/rest-api';
+
 import { authStateKey } from '../constants';
+import { useFronteggStore } from '../auth/mapAuthState';
 
-const getEntitlementsState = () => {
+/**
+ * @returns user state 
+ */
+const useGetUserState = () => {
   const authState = inject(authStateKey) as AuthState;
-  return authState.user?.entitlements as any;
+  return authState.user;
+};
+
+/**
+ * @returns true when need to use entitlements V2 API
+ */
+const useIsV2API = () => {
+  // TODO: check should it be immediate or we can recursively inject (check when app name change?)
+  const { appName } = useFronteggStore();
+  const [useEntitlementsV2] = FeatureFlags.getFeatureFlags([USE_ENTITLEMENTS_V2_ENDPOINT_FF], appName);
+  return useEntitlementsV2;
+};
+
+/**
+ * @param customAttributes consumer attributes
+ * @returns is entitled query data including: entitltments state, final attributes (consumer and frontegg) and API version to use
+ */
+const useEntitlementsQueryData = (customAttributes?: CustomAttributes) => {
+  const user = useGetUserState();
+  const entitlements = user?.entitlements;
+  const isV2 = useIsV2API();
+
+  const attributes = {
+    custom: customAttributes,
+    frontegg: user,
+  } as Attributes | undefined;
+
+  return {
+    entitlements,
```

Comment:
we can also access `entitlements` via `attributes.frontegg.entitlements` right?

Reply:
via auth.user.entitlements but I used a casting due to typing shit - that's why I decided to work with an hook